lib/management/spnego_krb5.c

```diff
@@ -112,12 +112,11 @@ static int parse_service_full_name(char *service_full_name,
 	}
 	*host_name = strndup(name, delim - name);
 	if (*host_name == NULL) {
-		free(*service_name);
```

Comment:
Can you explain why service_name should be freed here?

Reply:
If you meant shouldn’t it’s because it gets freed elsewhere.

Reply:
https://github.com/cifsd-team/ksmbd-tools/blob/406594f9ab9889d67c9c6c21e2792532d9a63b05/lib/management/spnego_krb5.c#L172

Reply:
Yes, But this function should free it on error. The error handling of this function seems to need refactoring. But if you fix it simply, you can add *service_name = NULL; here.